src/main/scala/com/google/cloud/spark/bigquery/direct/DirectBigQueryRelation.scala

```diff
@@ -127,23 +150,90 @@ private[bigquery] class DirectBigQueryRelation(
         prunedSchema,
         requiredColumns,
         options,
-        getClient).asInstanceOf[RDD[Row]]
+        getClient,
+        bigQueryClient).asInstanceOf[RDD[Row]]
 
     } finally {
       // scanTable returns immediately not after the actual data is read.
       client.close()
     }
   }
 
+  def getActualTable(
+      requiredColumns: Array[String],
+      filters: Array[Filter]
+    ): TableInfo = {
+    val tableDefinition = table.getDefinition[TableDefinition]
+    val tableType = tableDefinition.getType
+    if(options.viewsEnabled && TableDefinition.Type.VIEW == tableType) {
+      // get it from the view
+      val querySql = createSql(tableDefinition.getSchema, requiredColumns, filters)
+      log.debug(s"querySql is $querySql")
+      destinationTableCache.get(querySql, DestinationTableBuilder(querySql))
+    } else {
+      // use the default one
+      table
+    }
+  }
+
+  def createTableFromQuery(querySql: String): TableInfo = {
+    val destinationTable = createDestinationTable
```

Comment:
How are these temporary tables deleted after creation?

Reply:
I thought that it was set to 24 hours automatically... The connector now updates the destination table's expiration time immediately after it is created

Reply:
See https://github.com/GoogleCloudPlatform/spark-bigquery-connector/pull/83/commits/bdabb4216ea95818bf43569ad7d2e53e61d24c84